buildchain/buildchain/constants.py

```diff
@@ -93,4 +99,22 @@ def load_version_information() -> None:
 SHORT_VERSION : str = '{}.{}'.format(VERSION_MAJOR, VERSION_MINOR)
 VERSION : str = '{}.{}{}'.format(SHORT_VERSION, VERSION_PATCH, VERSION_SUFFIX)
 
+
+def git_ref() -> Optional[str]:
+    """Load version information from Git."""
+    ref = None
```

Comment:
`ref = None` is useless here.
Also, you could return the result of `strip` directly.

Could be written:

```suggestion
def git_ref() -> Optional[str]:
    """Return the current git revision.

    Return None if we can't get the information.
    """
    cmd = [config.GIT, 'describe']
    cmd.extend(['--always', '--long', '--tags','--dirty', '--broken'])
    try:
        stdout : bytes = subprocess.check_output(cmd)
        return stdout.decode('utf-8').rstrip()
    except subprocess.CalledProcessError:
        return None
```